### DISCUSSION TITLE
The next major version should have a desktop and freely movable and changeable windows

### DISCUSSION THREAD

**Discussion by hindermath** [tui-cs/Terminal.Gui#1116] [Ideas]
Wouldn't it be g'reat if Terminal.GUI, for example, had a desktop and windows in version 2.x that could overlap and move freely on the desktop. I refer to Turbo Vision by Borland from the 1980s/90s as a model for my proposal. This was the best TUI Framework I've worked with so far. Terminal.GUI is well on its way to running - and in my opinion, if it still had this capability - with the plus of the .Net 5 framework running on Windows, macOS and Linux!
![53F23BC7-A23B-473A-B829-1597E93054E3](https://user-images.githubusercontent.com/681882/109414211-0762ff00-79b2-11eb-9972-78c266bcd363.jpeg)
![D1177218-13D4-4016-8241-133661BA776D](https://user-images.githubusercontent.com/681882/109414212-07fb9580-79b2-11eb-9179-b57b89ba6337.jpeg)

**Comment by migueldeicaza** [comment]
Well, you could add the desktop view a view that did that.

The controls to close and other bits could be added.

**Comment by BDisp** [comment]
My PR #1331, with `Mdi Container` could be a start for this.

**Reply by hindermath** [reply]
That sounds good. I have a look on it!

**Comment by zherar7ordoya** [comment]
If you manage to emulate Borland's interfaces, this project will become legendary.

**Reply by tig** [reply]
See the Arrangement Scenario.  Choose the TurboPascal theme first.

**Reply by tig** [reply]
![image](https://github.com/user-attachments/assets/44acd703-719b-4e72-8033-98d6d24b0c50)
